### PR TITLE
[CK_TILE][FMHA] Add FP8 support for batch_prefill kernel

### PR DESCRIPTION
**Pull request by poyenc**
### Summary
This PR adds FP8 (float8_e4m3) support to the batch_prefill FMHA kernel, enabling mixed-precision computation with FP8 inputs and BF16 outputs.

### Key Changes
- **FP8BF16 kernel mode**: Add support for FP8 input (Q, K, V) with BF16 output in batch_prefill operation
- **Per-tensor quantization**: Support descale parameters for FP8 dequantization in the `qr_async` pipeline
- **Extended kernel coverage**: Add FP8 kernel instances with logits soft-capping support

### Implementation Details
- Uses `qr_async` pipeline with FP8 tensor core operations
- Supports per-tensor quantization via descale parameters
- Output dtype restricted to BF16 for FP8 inputs (fp8bf16 mode)

### Testing
Validated against BF16 reference implementation with accuracy threshold < 0.055

## Checklist

Please put an `x` into the boxes that apply. You can also fill these out after creating the PR. If you're not sure, please don't hesitate to ask.

- [x] I have added tests relevant to the introduced functionality, and the unit tests are passing locally
- [ ] I have added the test to REGRESSION_TESTS list defined at the top of CMakeLists.txt in tests/CMakeLists.txt, **IF** the test takes more than 30 seconds to run.
- [ ] I have added inline documentation which enables the maintainers with understanding the motivation
- [ ] I have removed the stale documentation which is no longer relevant after this pull request
- [ ] (If this change is user-facing) I have added release notes which provide the end users with a brief summary of the improvement from this pull request
- [x] I have run `clang-format` on all changed files
- [x] Any dependent changes have been merged

## Discussion

If this is a relatively large or complex change, feel free to start a discussion by explaining why you chose the solution you did and what alternatives you considered

### CODE REVIEW

File: include/ck_tile/ops/fmha/kernel/fmha_fwd_kernel.hpp [hunk 1]
```diff
@@ -1499,14 +1499,28 @@ struct FmhaFwdKernel
 
             AttentionVariant variant;
             const auto variant_params = [&] {
+                const float scale_s = [&] {
+                    if constexpr(QScaleEnum == BlockAttentionQuantScaleEnum::PERTENSOR)
+                    {
+                        float q_descale = *(reinterpret_cast<const float*>(kargs.q_descale_ptr));
+                        float k_descale = *(reinterpret_cast<const float*>(kargs.k_descale_ptr));
+
+                        return kargs.scale_s * q_descale * k_descale;
```
Comment (ltqin) [inline include/ck_tile/ops/fmha/kernel/fmha_fwd_kernel.hpp]: Actually, the best method is to pass q_descale * k_descale to s_acc_ element_func. Will passing it to s_acc_ element_func affect performance?

File: include/ck_tile/ops/fmha/kernel/fmha_fwd_kernel.hpp [hunk 2]
```diff
@@ -1499,14 +1499,28 @@ struct FmhaFwdKernel
 
             AttentionVariant variant;
             const auto variant_params = [&] {
+                const float scale_s = [&] {
+                    if constexpr(QScaleEnum == BlockAttentionQuantScaleEnum::PERTENSOR)
```
Comment (ltqin) [inline include/ck_tile/ops/fmha/kernel/fmha_fwd_kernel.hpp:1503]: move scale_s into   if constexpr(kHasLogitsSoftCap) {}?

Comment (poyenc) [reply include/ck_tile/ops/fmha/kernel/fmha_fwd_kernel.hpp:1503]: Yes, we will need to re-compute scale_s in variant_params

**Review by ltqin — DISMISSED** [review]

**Review by asleepzzz — DISMISSED** [review]

**Review by spolifroni-amd — DISMISSED** [review]

**Review by asleepzzz — APPROVED** [review]
